pennylane/estimator/compact_hamiltonian.py

```diff
@@ -178,3 +180,194 @@ def __post_init__(self):
+    to obtain resource estimates. Consider for example the Hamiltonian:
+
+    .. math::
+
+        \hat{H} = 0.1 \cdot \Sigma^{30}_{j=1} \hat{X}_{j} \hat{X}_{j+1}
+        - 0.05 \cdot \Sigma^{30}_{k=1} \hat{Y}_{k} \hat{Y}_{k+1} + 0.25 \cdot \Sigma^{40}_{l=1} \hat{X}_{l}
+
+    This Hamiltonian is represented in a compact form using ``PauliHamiltonian``:
+
+    >>> import pennylane.estimator as qre
+    >>> pauli_ham = qre.PauliHamiltonian(
+    ...     num_qubits = 40,
+    ...     pauli_terms = {"X":40, "XX":30, "YY":30},
+    ...     one_norm = 14.5,  # (|0.1| * 30) + (|-0.05| * 30) + (|0.25| * 40)
+    ... )
+    >>> pauli_ham
+    PauliHamiltonian(num_qubits=40, one_norm=14.5, pauli_terms={'X': 40, 'XX': 30, 'YY': 30})
+
+    The Hamiltonian can be used as input for other subroutines, like
+    :class:`~.estimator.templates.trotter.TrotterPauli`:
+
+    >>> num_steps, order = (10, 2)
+    >>> res = qre.estimate(qre.TrotterPauli(pauli_ham, num_steps, order))
+    >>> print(res)
+    --- Resources: ---
+     Total wires: 40
+       algorithmic wires: 40
+       allocated wires: 0
+         zero state: 0
+         any state: 0
+     Total gates : 9.880E+4
+       'T': 8.800E+4,
+       'CNOT': 2.400E+3,
+       'Z': 1.200E+3,
+       'S': 2.400E+3,
+       'Hadamard': 4.800E+3
+
+    .. details::
+        :title: Usage Details
+
+        The terms of the Hamiltonian can also be separated into groups such that all operators in
+        the group commute. Users can instantiate the ``PauliHamiltonian`` by specifying these
+        groups of terms directly.
+
+        >>> import pennylane.estimator as qre
+        >>> commuting_groups = (
+        ...     {"X": 40, "XX": 30}, # first commuting group
+        ...     {"YY": 30}, # second commuting group
+        ... )
+        >>> pauli_ham = qre.PauliHamiltonian(
+        ...     num_qubits = 40,
+        ...     pauli_terms = commuting_groups,
+        ...     one_norm = 14.5,  # (|0.1| * 30) + (|-0.05| * 30) + (|0.25| * 40)
+        ... )
+        >>> pauli_ham
+        PauliHamiltonian(num_qubits=40, one_norm=14.5, pauli_terms=({'X': 40, 'XX': 30}, {'YY': 30}))
+
+        Note that providing more information will generally lead to more accurate resource estimates.
+
+        >>> num_steps, order = (10, 2)
+        >>> res = qre.estimate(qre.TrotterPauli(pauli_ham, num_steps, order))
+        >>> print(res)
+        --- Resources: ---
+         Total wires: 40
+           algorithmic wires: 40
+           allocated wires: 0
+             zero state: 0
+             any state: 0
+         Total gates : 5.266E+4
+           'T': 4.708E+4,
+           'CNOT': 1.260E+3,
+           'Z': 600,
+           'S': 1.200E+3,
+           'Hadamard': 2.520E+3
+
+    """
+
+    def __init__(
+        self,
+        num_qubits: int,
+        pauli_terms: dict | Iterable[dict],
+        one_norm: float | None = None,
```

Comment:
Doesn't match the args in the docsstring.
```suggestion
        one_norm: float | None = None,
```